code/Gerrit.md

```diff
@@ -0,0 +1,282 @@
+
+One of the most common issues people have with Gerrit is getting used to an unfamiliar git workflow. Once you get some practice with it, you will be able to decide whether or not you prefer it. The advantage of the different git workflow is that it requires you to be very mindful of your git history, and organize your code into sensible commits. The disadvantage of the different git workflow is that it requires you to be very mindful of your git history, and organize your code into sensible commits. 
+
+## Common git+gerrit issues and how to fix them
+
+### Errors trying to push to Gerrit
+
+If you get an error submitting a change to Gerrit, the first thing you should check is that you have pulled the latest commits from the `main` branch:
+
+```
+git pull --rebase origin main
+git push gerrit HEAD:refs/for/main
+```
+
+You may also see an error if you are pushing a change that is identical to one that has already been pushed to Gerrit. If you know this, and want to push anyway (e.g. go back to a previous version of the change), then simply run the following to update the commit's timestamp:
+
+```
+git commit --amend --no-edit
+git push gerrit HEAD:refs/for/main
+```
+
+### Dealing with merge conflicts
+
+Sometimes, when you do `git pull --rebase origin main`, you will see a message about conflicting files that you will need to resolve. Getting specific about how to fix conflicts in git is out of the scope of this document, but one important thing to keep in mind is that pulling `main` with `--rebase` basically temporarily removes your commits, puts all the new commits from the `main` branch in your git history, then puts your commits back on top. This way, your commits _always_ remain as the latest commits in the git history. 
+
+What this means is that the conflicts you see will look reversed from what you may be used to. If you are used to doing `git pull origin main`, your conflicts would look like this:
+
+```
+<<<<<<< HEAD
+conflicting changes from main
+=======
+conflicting changes from your-local-branch
+>>>>>>>
+```
+
+But with `--rebase`, the same conflict looks like:
+
+```
+<<<<<<< HEAD
+conflicting changes from your-local-branch
+=======
+conflicting changes from main
+>>>>>>>
+```
+
+### Updating a multi-commit change
+
+Let's say you are working on a big change that requires three separate commits: A, B, and C. Your local branch might look like this:
+
+* C commit
+* B commit
+* A commit
+* latest commit on `main`
+* ...
+
+Now, someone reviews your B change, and asks you to update it. How do you do this? `git commit --amend` will update the latest commit, "C commit", but you need to keep your changes for B and C separate.
+
+There are many ways to approach this problem, but one way is to do an interactive git rebase. This will look like:
```

Comment:
Perhaps we can drop this link here https://git-scm.com/docs/git-rebase#_interactive_mode